catchup/service.go

```diff
@@ -198,7 +198,14 @@ func (s *Service) fetchAndWrite(fetcher rpcs.Fetcher, r basics.Round, prevFetchC
 
 		// Check that the block's contents match the block header (necessary with an untrusted block because b.Hash() only hashes the header)
 		if !block.ContentsMatchHeader() {
-			s.log.Warnf("fetchAndWrite(%v): block contents do not match header (attempt %d)", r, i)
+			// Check if this mismatch is due to an unsupported protocol version
+			if _, ok := config.Consensus[block.BlockHeader.CurrentProtocol]; !ok {
+				if i == 10 {
+					s.log.Errorf("fetchAndWrite(%v): unsupported protocol version detected", )
+				}
```

Comment:
If you're adding a special handling for mismatching protocol version - 
1. Given that we *agreed* upon this protocol upgrade, there is no reason to wait for the 10th attempt before reporting it. You can report it right away.
2. Once you're reported this case, call 'client.Close()' and return false. There is absolutely no reason to make another attempt for the naive case.
3. For the non-naive case where you've received this invalid response from a hostile relay, the existing logic would retry again anyway, but with some delay. That would protect us against draining network resources endlessly once we cannot move forward due to protocol mismatch.

Reply:
great! thanks I'll make this change.